### PR TITLE
`Paywalls`: replaced submodule with `gitignore`d reference

### PR DESCRIPTION
**Pull request by NachoSoto**
The `submodule` approach had 2 issues:
- Inconsistent results if users don't have a valid SSH key (#3124)
- `SPM` actually clones the submodule, which users don't need

This new approach only fetches the snapshots through a new `fastlane fetch_snapshots` lane.

#3116 will be updated to support this.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -757,6 +760,38 @@ platform :ios do
...
+
+        sh "git", "clone", "git@github.com:RevenueCat/#{snapshots_repo_name}.git"
+      end
+
+      commit = current_snapshots_commit
+      UI.message("Updating snapshots repo to #{commit}")
+      Dir.chdir(snapshots_repo_name) do
+        sh "git", "fetch", "origin"
+        sh "git", "checkout", commit
+      end
```
Comment (joshdholtz) [inline fastlane/Fastfile:777]: Do we need to check if the repo is dirty at error out before this? 🤔 

Or do we need to do a hard reset so its clean?

**Review by joshdholtz — APPROVED** [review]
One question about if we need to check if its dirty before updating 🤷‍♂️  But not a blocker! We can keep iterating 😊

File: fastlane/Fastfile (hunk 1 above)
Comment (NachoSoto) [reply fastlane/Fastfile:777]: Good point that would make it more resilient.

Comment (NachoSoto) [reply fastlane/Fastfile:777]: Actually I'm gonna leave it like this. If it's dirty then you want to handle that error so we don't make you lose whatever local changes you have.

**Pull request merged**